File: Android/Score java/app/src/main/java/it/unisannio/cp/orange/scorejava/activity/SettingsActivity.java
package it.unisannio.cp.orange.scorejava.activity;


/*
 *  Author: Raffaele Mignone
 *  Mat: 863/747
 *  Date: 28/11/17
 *
 */


import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.CompoundButton;
import android.widget.Switch;
import android.widget.TextView;

import it.unisannio.cp.orange.scorejava.R;

public class SettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        final SharedPreferences sp = getSharedPreferences("settings", Context.MODE_PRIVATE);

        final TextView pro = findViewById(R.id.pro);
        if(sp.getBoolean("pro", false))             //determina se l'utente ha comprato la versione premium
            pro.setText(R.string.settings_pro);
        else
            pro.setText(R.string.settings_became_pro);

        pro.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {                    //metodo di test per vedere il cambio tra pro e non-pro
                SharedPreferences.Editor editor = sp.edit();
                boolean not = !sp.getBoolean("pro", false);
                editor.putBoolean("pro", not);
                editor.apply();
                if(not)
                    pro.setText(R.string.settings_pro);
                else
                    pro.setText(R.string.settings_became_pro);
                return false;
            }
        });

        TextView github = findViewById(R.id.github);
        github.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {           //crea un intent per visualizzare un sito web
                Intent web = new Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.settings_github_uri)));
                startActivity(web);
            }
        });

        Switch order = findViewById(R.id.order);
        order.setChecked(sp.getInt("order", 1)==1);         //imposta lo switch nella corretta posizione
        order.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {    //salva sulle sp la nuova preferenza dell'ordine
                SharedPreferences.Editor editor = sp.edit();
                if(isChecked)
                    editor.putInt("order", 1);
                else
                    editor.putInt("order", -1);
                editor.apply();
            }
        });

    }
}

File: Android/Score30/app/src/main/java/it/unisannio/cp/orange/score30/activity/SettingsActivity.kt
/*
 *      Copyright (c) 2017 Raffaele Mignone
 *
 *      This file is part of  Score30
 *
 *      Score30 is free software: you can redistribute it and/or modify
 *      it under the terms of the GNU General Public License as published by
 *      the Free Software Foundation, either version 2 of the License, or
 *      (at your option) any later version.
 *
 *      Score30 is distributed in the hope that it will be useful,
 *      but WITHOUT ANY WARRANTY; without even the implied warranty of
 *      MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *      GNU General Public License for more details.
 *
 *      You should have received a copy of the GNU General Public License
 *      along with Score30.  If not, see <http://www.gnu.org/licenses/>.
 */

package it.unisannio.cp.orange.score30.activity

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.net.Uri
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import it.unisannio.cp.orange.score30.R
import kotlinx.android.synthetic.main.activity_info.*

class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_info)
        val sp = getSharedPreferences("settings", Context.MODE_PRIVATE)

        if(sp.getBoolean("pro", false))
            pro.setText(R.string.settings_pro)
        else
            pro.setText(R.string.settings_became_pro)

        pro.setOnLongClickListener({ _ ->
            val editor = sp.edit()
            val b = sp.getBoolean("pro", false)
            editor.putBoolean("pro", !b)
            editor.apply()
            if(b)
                pro.setText(R.string.settings_became_pro)
            else
                pro.setText(R.string.settings_pro)
            true
        })

        github.setText(R.string.settings_github)
        github.setOnClickListener({ _ ->
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.settings_github_uri)))
            startActivity(intent)
        })
        order.isChecked = sp.getInt("order", 1) == 1

        order.setOnCheckedChangeListener({ _, isChecked: Boolean ->
            val editor = sp.edit()
            if(isChecked)
                editor.putInt("order", 1)
            else
                editor.putInt("order", -1)
            editor.apply()
        })
    }
}
